File: add/codesnippet/CPP/m-system.windows.forms.g_0_1.cpp
private:
   void Clear_Clicked( Object^ /*sender*/, System::EventArgs^ /*e*/ )
   {
      // TablesAlreadyAdded set to false so that table styles can be added again.
      TablesAlreadyAdded = false;
      myLabel->Text = "All Table Styles Cleared.";

      // Clear all the column styles and also table style for the grid.
      IEnumerator^ myEnum = myDataGrid->TableStyles->GetEnumerator();
      while ( myEnum->MoveNext() )
      {
         DataGridTableStyle^ myTableStyle = safe_cast<DataGridTableStyle^>(myEnum->Current);
         GridColumnStylesCollection^ myColumns = myTableStyle->GridColumnStyles;
         myColumns->Clear();
      }

      myDataGrid->TableStyles->Clear();
   }

File: add/codesnippet/CSharp/m-system.windows.forms.g_0_1.cs
   private void Clear_Clicked(object sender, System.EventArgs e)
   {
      // TablesAlreadyAdded set to false so that table styles can be added again.
      TablesAlreadyAdded = false;
      myLabel.Text = "All Table Styles Cleared.";
      // Clear all the column styles and also table style for the grid.
      foreach (DataGridTableStyle myTableStyle in myDataGrid.TableStyles)
      {
         GridColumnStylesCollection myColumns = myTableStyle.GridColumnStyles;
         myColumns.Clear();
      }
      myDataGrid.TableStyles.Clear();
   }